nncf/torch/model_creation.py

```diff
@@ -171,34 +170,26 @@ def create_nncf_network(model,
             "building, then the wrap_inputs_fn parameter MUST also be specified and be consistent with "
             "the input wrapping done in dummy_forward_fn.")
 
-    # Compress model that will be deployed for the inference on target device. No need to compress parts of the
-    # model that are used on training stage only (e.g. AuxLogits of Inception-v3 model) or unused modules with weights.
-    # As a consequence, no need to care about spoiling BN statistics, as they're disabled in eval mode.
-    model.eval()
-
-    input_info_list = create_input_infos(config)
-    scopes_without_shape_matching = config.get('scopes_without_shape_matching', [])
-    ignored_scopes = config.get('ignored_scopes')
-    target_scopes = config.get('target_scopes')
-
-    original_model_accuracy = None
-    if is_accuracy_aware_training(config):
-        if config.has_extra_struct(ModelEvaluationArgs):
-            evaluation_args = config.get_extra_struct(ModelEvaluationArgs)
-            with torch.no_grad():
-                original_model_accuracy = evaluation_args.eval_fn(model)
-                nncf_logger.info(f"Uncompressed model accuracy = {original_model_accuracy}")
-
-    nncf_network = NNCFNetwork(model, input_infos=input_info_list,
-                               dummy_forward_fn=dummy_forward_fn,
-                               wrap_inputs_fn=wrap_inputs_fn,
-                               wrap_outputs_fn=wrap_outputs_fn,
-                               ignored_scopes=ignored_scopes,
-                               target_scopes=target_scopes,
-                               scopes_without_shape_matching=scopes_without_shape_matching,
-                               original_model_accuracy=original_model_accuracy)
-
-    nncf_network.get_tracing_context().disable_trace_dynamic_graph()
+    # Preserve `.training`/`.requires_grad` state since we will be building NNCFNetwork in `.eval` mode
+    with training_mode_switcher(model, is_training=False):
```

Comment:
@ljaljushkin please comment whether or not this impacts NAS approach. Maybe we need to extend NNCFNetwork with an option to build train-mode or eval-mode graphs based on a parameter, if NAS needs the training-mode graph to work.

Reply:
It shouldn't affect the scope of working models. 
Potentially, the training-mode graph should help to resolve the issue with Aux branches in the Inception-v3 model. 
ticket 60976 (https://github.com/openvinotoolkit/nncf/pull/867)